activitychart-basic.html

```diff
@@ -0,0 +1,274 @@
+        accentcolor: {
+          type: String,
+          value: '#F5E8BB',
+        },
+        /** Number of colored dots per column (opt, default 8) */
+        squaresPerColumn: {
+          type: Number,
+          value: 8,
+        },
+        /** Width of square (opt, default: 15) */
+        squareWidth: {
+          type: Number,
+          value: 15,
+        },
+        /** Spacing of squares (opt, default: 20) */
+        squareSpacing: {
+          type: Number,
+          value: 20,
+        },
+        //this could also be solved by setting value to a function calculating the default min/max value from the dataset (but only if we receive the dataset as a JSON object isntead of just receiving a URL)
+        /** Minimum value from the dataset that is used for the color scale can be overwritten. */
+        minValue: {
+          type: Number,
+          value: null
+        },
+        /** Maximum value from the dataset that is used for the color scale can be overwritten. */
+        maxValue: {
+          type: Number,
+          value: null
+        },
+        /** Labels for x and y axis.
+        Please provide as {'x': {'label': String}, 'y': {'label': String, 'values': String[]}}
+        The correct labels for the x axis are chosen programmatically based on the data.*/
+        axisLabels: {
+          type: Object,
+          value: {
+            "x": {
+              "label": "Tag"
+            },
+            "y": {
+          		"label": "Stunde",
+              "values": ["00:00", "01:00", "02:00", "03:00", "04:00", "05:00", "06:00", "07:00", "08:00", "09:00", "10:00", "11:00", "12:00", "13:00", "14:00", "15:00", "16:00", "17:00", "18:00", "19:00", "20:00", "21:00", "22:00", "23:00"]
+            }
+          }
+        }
+      },
+
+      attached: function() {
+        if(this.dataurl != "") {
+          this.$.ajaxCaller.generateRequest();
+        }
+        else {
+          this._plotDiagram();
+        }
+      },
+      /**
+      * Transforms the transmitted data into a plot
+      *
+      * @param {Object} data The data to be plotted
+      */
+      handleResponse: function (data) {
+        var goodData = this._normalizeJSON(data.detail.response);
+        this.dataset = goodData[0]["y"];
+        this.labels = goodData[0]["x"];
+        this._plotDiagram();
+      },
+
+      _buildDataset: function(labels, dataset) {
+        // create data like this:
+        // [ {'x': timeStamp0, 'y': value0}, ... ,{'x': timeStampN, 'y': valueN}]
+        var data = [];
+        for(var j = 0; j < labels.length; j++) {
+          data[j] = {'x': labels[j], 'y': dataset[j]};
+        }
+        return data;
+      },
+
+      _plotDiagram: function () {
+        console.log("Plotting...");
+        console.log(this.dataset);
+        console.log(this.labels);
+        var diagramDiv = this.$.diagram;
+
+        var data = this._buildDataset(this.labels, this.dataset);
+
+        var dataset = this.dataset;
```

Comment:
if we move this line above `var data = this._buildDataset(this.labels, this.dataset);` we can omit the `this.` in the latter part of the params